### PR TITLE
Controlled DatePicker value, Fixes #4630

### LINKED ISSUE #4630: [DatePicker] controlled component
_I am not sure if this is a bug or a feature request so i explain it with the bug template anyway_

### Bug Report

- __Package version(s)__: tested >= 5.3.0
- __Browser and OS versions__: MacOS Chrome 65, (I think also all others)

#### Describe the issue:

When i pass a value to the DatePicker component it renders correctly. When I then click on the DatePicker to select a other date and its not allowed by any reason and its not set as value for the DatePicker the Picker renders it anyway.

#### Actual behavior:

It renders its internal state beside if the given value is different.

#### Expected behavior:

I thought when I pass a value it should render this value and only showing the value and not its internal state.

### If applicable, please provide a codepen repro:

[Edit v0kywjv207](https://codesandbox.io/s/v0kywjv207)

in the repo i only log the selected date, but doesn't set it to the component state. The component displays it anyway. Its got weirder if I set a harcoded date with setState.

### PR DESCRIPTION
**Pull request by lrecknagel**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #4630
- [x] Include a change request file using `$ npm run change`

#### Description of changes

As I described in the issues, mentioned above, the component update its internal state of the current date value always. This shouldn´t be happen, when a value is passed and this values does not update.

In _DatePicker.base.tsx_ I removed the always applied state update for _selectedDate_ and _formattedDate_.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by Vitalius1** [comment]
@lrecknagel you would need to run `rush change` to generate a change log.

**Comment by KevinTCoughlin** [comment]
Thanks for the fix @lrecknagel! It'd be awesome if we could add a unit test for the bug this is addressing as part of this PR.

**Comment by lrecknagel** [comment]
@KevinTCoughlin sorry for the late response. I try to provide the tests this week 👍

**Comment by KevinTCoughlin** [comment]
> 
> 
> @KevinTCoughlin sorry for the late response. I try to provide the tests this week 👍

Fantastic! Thank you so much. If you get stuck let me know and I can help unblock. I would have written the test myself, but I didn't quite understand the bug hence my ask 😆.

**Comment by mikewheaton** [comment]
@lorejoh12 As code owner, could you review this change?

**Comment by lrecknagel** [comment]
Finally wrote the tests. Please tell me about concerns. I am really new to this stuff.

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.base.tsx [hunk 1]
```diff
@@ -261,17 +261,12 @@ export class DatePickerBase extends BaseComponent<IDatePickerProps, IDatePickerS
...
+    const { onSelectDate } = this.props;
 
     if (this.props.calendarProps && this.props.calendarProps.onSelectDate) {
       this.props.calendarProps.onSelectDate(date);
     }
 
-    this.setState({
-      selectedDate: date,
-      formattedDate: formatDate && date ? formatDate(date) : ''
-    });
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.base.tsx]: Are you sure this should be removed entirely? What about uncontrolled scenarios? It appears [this has broken the existing uncontrolled examples](http://fabricweb.z5.web.core.windows.net/pr/refs/pull/6590/merge/#/examples/datepicker).

Comment (lrecknagel) [reply packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.base.tsx]: I will take care of this.

**Comment by dzearing** [comment]
Ping on this. If you try the deploy link, it is pretty busted (can't change dates.)

The build also failed because you're changing the public contract of a package (`@uifabric/styling`).

@uifabric/styling (16.37 seconds)
You have changed the public API signature for this project. Please overwrite etc/styling.api.ts with a copy of temp/styling.api.ts and then request an API review. See the Git repository README.md for more info.

I need to look at what's up...

**Comment by aneeshack4** [comment]
@lrecknagel Hello! I see that this PR has been open for quite a while - are you planning on getting back to this & addressing the review comments & the broken build? If there are any questions we can help address, please let us know!

**Comment by micahgodbolt** [comment]
@lorejoh12 is this something you're interested in having in this control? Do you want to pick it up or close this down?

**Comment by lrecknagel** [comment]
* First: Sorry leave this conversation for so long.
* I updated the PR and from my point of view, now my issue is fixed and the uncontrolled components work.

**Comment by cliffkoh** [comment]
```
/packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.test.tsx:301:9
ERROR: 301:9  no-use-before-declare  variable 'datePicker' used before declaration

[office-ui-fabric-react 11:51:52 PM] Error: tslint (38.667s): Error: Command failed: node \scripts\node_modules\tslint\lib\tslintCli.js --project C:\git\oufr-tmp\packages\office-ui-fabric-react\tsconfig.json -t stylish -r \scripts\node_modules\tslint-microsoft-contrib

src/components/DatePicker/DatePicker.test.tsx:294:26 - error TS2339: Property '_calendar' does not exist on type 'Component<{}, {}, any>'.

294       datePickerInstance._calendar.current._dayPicker.current._onSelectDate(otherValue);
                             ~~~~~~~~~


src/components/DatePicker/DatePicker.test.tsx:319:26 - error TS2339: Property '_calendar' does not exist on type 'Component<{}, {}, any>'.

319       datePickerInstance._calendar.current._dayPicker.current._onSelectDate(otherValue);
                             ~~~~~~~~~



[office-ui-fabric-react 11:51:53 PM] Error: ts (39.523s): Error: Command failed: node \common\temp\node_modules\.registry.npmjs.org\typescript\2.8.4\node_modules\typescript\lib\tsc.js -outDir lib -t es5 -m es2015 --pretty
```

**Pull request closed without merging**